ext/ScreenShot/ScreenShot.h

```diff
@@ -147,16 +147,13 @@ namespace irr
 				auto destinationBuffer = driver->createGPUBufferOnDedMem(memoryRequirements);
 
 				auto fence = downloadImageMipLevel(driver, gpuImage.get(), destinationBuffer.get());
-				while (fence->waitCPU(1000ull, fence->canDeferredFlush()) == video::EDFR_TIMEOUT_EXPIRED) {}
 
 				auto destinationBoundMemory = destinationBuffer->getBoundMemory();
 				destinationBoundMemory->mapMemoryRange(video::IDriverMemoryAllocation::EMCAF_READ, { 0u, memoryRequirements.vulkanReqs.size });
 				auto texelBuffer = core::make_smart_refctd_ptr<asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t>>>(memoryRequirements.vulkanReqs.size, destinationBoundMemory->getMappedPointer(), core::adopt_memory);
 				
 				auto mapPointerGetterFence = driver->placeFence(true);
 				while (mapPointerGetterFence->waitCPU(1000ull, mapPointerGetterFence->canDeferredFlush()) == video::EDFR_TIMEOUT_EXPIRED) {}
```

Comment:
you place the fence as soon as possible (line 149) but you wait as late as possible (line 178 just before tryToWrite)

Also you have an unused variable/fence called `fence`

Reply:
did you move them?

Reply:
yep, in upcoming commit